code/client/cl_main.c

```diff
@@ -4708,3 +4714,90 @@ qboolean CL_CDKeyValidate( const char *key, const char *checksum ) {
 	return qfalse;
 #endif
 }
+
+/*
+==================
+CL_RegisterProtocolHandler_f
+==================
+*/
+void CL_RegisterProtocolHandler_f( void )
+{
+#ifndef PROTOCOL_HANDLER
+	Com_Printf( "Not built with protocol handler support.\n" );
+#elif _WIN32
+
+	HKEY key;
+	int error;
+	char *path;
+	char value[MAX_PATH];
+
+	error = RegCreateKeyEx( HKEY_CURRENT_USER, "Software\\Classes\\" PROTOCOL_HANDLER, 0, NULL, 0, KEY_SET_VALUE, NULL, &key, NULL );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error opening registry key.\n" );
+		return;
+	}
+
+	path = Sys_ExecutablePath( );
+	error = RegSetValueEx( key, "CustomUrlApplication", 0, REG_SZ, (LPBYTE) path, strlen( path ) + 1 );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error writing to registry.\n" );
+		return;
+	}
+
+	strcpy( value, "\"%1\"" );
+	error = RegSetValueEx( key, "CustomUrlArguments", 0, REG_SZ, (LPBYTE) value, strlen( value ) + 1 );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error writing to registry key.\n" );
+		return;
+	}
+
+	strcpy( value, "" );
+	error = RegSetValueEx( key, "URL Protocol", 0, REG_SZ, (LPBYTE) value, strlen( value ) + 1 );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error writing to registry.\n" );
+		return;
+	}
+	RegCloseKey( key );
+
+	error = RegCreateKeyEx( HKEY_CURRENT_USER, "Software\\Classes\\" PROTOCOL_HANDLER "\\DefaultIcon", 0, NULL, 0, KEY_SET_VALUE, NULL, &key, NULL );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error opening registry key.\n" );
+		return;
+	}
+
+	snprintf( value, sizeof(value), "%s,0", path );
+	error = RegSetValueEx( key, "", 0, REG_SZ, (LPBYTE) value, strlen( value ) + 1 );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error writing to registry.\n" );
+		return;
+	}
+	RegCloseKey( key );
+
+	error = RegCreateKeyEx( HKEY_CURRENT_USER, "Software\\Classes\\" PROTOCOL_HANDLER "\\shell\\open\\command", 0, NULL, 0, KEY_SET_VALUE, NULL, &key, NULL );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error opening registry key.\n" );
+		return;
+	}
+
+	snprintf( value, sizeof(value), "\"%s\" --uri \"%%1\"", path );
+	error = RegSetValueEx( key, "", 0, REG_SZ, (LPBYTE) value, strlen( value ) + 1 );
+	if ( error != ERROR_SUCCESS )
+	{
+		Com_Printf( "Error writing to registry.\n" );
+		return;
+	}
+	RegCloseKey( key );
+
+	Com_Printf( "Registered protocol " PROTOCOL_HANDLER ":// at HKEY_CURRENT_USER\\Software\\Classes\\" PROTOCOL_HANDLER "\\\n" );
+
+#else
+	Com_Printf( "Only implemented for Windows.\n" );
```

Comment:
A context should be given here in this message - like protocol handler registration is only implemented for windows.

if this message is needed at all.